tutorials/llm/slimpajama/pretraining.ipynb

```diff
@@ -0,0 +1,171 @@
+    "        seq_length=seq_length,\n",
+    "        global_batch_size=gbs,\n",
+    "        micro_batch_size=mbs,\n",
+    "        tokenizer=run.Config(SentencePieceTokenizer, model_path=\"/data/tokenizer/tokenizer.model\"),\n",
+    "        split=\"99,8,2\",\n",
+    "        num_workers=2,\n",
+    "        index_mapping_dir=\"/data/index_mapping\",\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Configuring the recipe and launching pretraining\n",
+    "Once the data module is defined, you can use an existing recipe and replace the data module as shown below.\n",
+    "To learn more about the recipes, refer to the [quickstart](https://docs.nvidia.com/nemo-framework/user-guide/latest/nemo-2.0/quickstart.html)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def configure_recipe(nodes: int = 1, gpus_per_node: int = 1):\n",
+    "    recipe = llm.llama3_8b.pretrain_recipe(\n",
+    "        dir=\"/checkpoints/llama-new\", # Path to store checkpoints\n",
+    "        name=\"llama_pretraining\",\n",
+    "        num_nodes=nodes,\n",
+    "        num_gpus_per_node=gpus_per_node,\n",
+    "    )\n",
+    "\n",
+    "    recipe.model.config.num_layers = 1\n",
+    "    recipe.model.config.hidden_size = 128\n",
+    "    recipe.trainer.max_steps = 30\n",
+    "    recipe.data = slimpajama(\n",
+    "        gbs=32,\n",
+    "        mbs=1,\n",
+    "    )\n",
+    "    recipe.trainer.val_check_interval = 20\n",
+    "    recipe.trainer.strategy.context_parallel_size = 1\n",
+    "    recipe.log.ckpt.save_optim_on_train_end = True\n",
+    "    return recipe"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def local_executor_torchrun(nodes: int = 1, devices: int = 1) -> run.LocalExecutor:\n",
+    "    # Env vars for jobs are configured here\n",
+    "    env_vars = {\n",
+    "        \"TORCH_NCCL_AVOID_RECORD_STREAMS\": \"1\",\n",
+    "        \"NEMO_ENV_VARNAME_TESTING\": \"1\",\n",
+    "        \"CUDA_VISIBLE_DEVICES\": \"0\"\n",
+    "    }\n",
+    "\n",
+    "    executor = run.LocalExecutor(ntasks_per_node=devices, launcher=\"torchrun\", env_vars=env_vars)\n",
+    "    return executor\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def run_pretraining():\n",
+    "    recipe = configure_recipe()\n",
+    "    executor = local_executor_torchrun(nodes=recipe.trainer.num_nodes, devices=recipe.trainer.devices)\n",
+    "\n",
+    "    run.run(recipe, executor=executor)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run pretraining\n",
+    "Now you can just call the `run_pretraining` function to start pretraining on your local machine using torchrun."
```

Comment:
fix punctuation

Now ,you can just call the `run_pretraining` function to start pretraining on your local machine using torchrun.